packages/@ember/-internals/glimmer/lib/helpers/array.ts

```diff
@@ -0,0 +1,38 @@
+import { Arguments, VM } from '@glimmer/runtime';
+
+/**
+@module ember
+*/
+
+/**
+   Use the `{{array}}` helper to create an array to pass as an option to your
+   components.
+
+   ```handlebars
+   {{my-component people=(array
+     'Tom Dade'
+     'Yehuda Katz'
+     this.myOtherPerson)
+   }}
+   ```
+
+   Would result in an object such as:
+
+   ```js
+   ['Tom Date', 'Yehuda Katz', this.get('myOtherPerson')]
+   ```
+
+   Where the 3rd item in the array is bound to updates of the `myOtherPerson` property.
+
+   @method array
+   @for Ember.Templates.helpers
+   @param {Array} options
+   @return {Array} Array
+   @category array-helper
+   @since 3.7.0
+   @public
+ */
+
+export default function(_vm: VM, args: Arguments) {
```

Comment:
Need return type

Reply:
Would that be `any[]`?

Reply:
It seems seem to like that change alone.

<img width="1060" alt="screen shot 2018-10-10 at 11 00 56 am" src="https://user-images.githubusercontent.com/230476/46756245-d5835d80-cc7b-11e8-9a11-483c302c9111.png">

The only way I found to make this work is by casting to any.

```ts
export default function(_vm: VM, args: Arguments): any[] {
  return args.positional.capture() as any;
}
```

Reply:
I believe it should be `PathReference<Opaque>`, or `PathReference<Opaque[]>` if we want to be precise

Reply:
Can you also fix `hash.ts` to have a return type of `PathReference<Opaque>`? (or `PathReference<Dict<Opaque>>`)

Reply:
Pushed the change. I will open a new PR for the return type of hash helper to keep it separated.